### ISSUE TITLE
Flag to not 'fix' issues XML document

### ISSUE THREAD

**Issue by paulhiggs** [chrisbottin/xml-formatter#58]
I use xmlFormatter to format an XML document before it is sent to other functions for syntax checking (a.la. xmllint) and semantioc checking (my own code).
Currently xmlFormatter seems to 'fix' problems with mismatching end tags, i.e.
````
import xmlFormat from 'xml-formatter';
console.log(xmlFormat('<root><content><p xml:space="preserve">This is <b>some</b> content.</contentXX></p>',
 {throwOnFailure:true));
````
yields
````
<root>
    <content>
        <p xml:space="preserve">This is <b>some</b> content.</p>
    </content>
</root>
````
The closing `</contentXX>` has been renamed and a closing `</root>` has been added.

In some applications, this is very helpful, but it makes it impossible to inform the application that the source XML is malformed. 
Can a flag be added just to 'pretty' the XML and not auto-correct it?
Also, in this case, shouldn't an exception be thrown?

**Comment by chrisbottin** [comment]
Hi @paulhiggs 

Thanks for submitting your issue.

Yes I agree, the XML parser might be a little too forgiving right now. I will see if I can make this a little stricter.

Right now it would fail parsing if the XML was `<root><content><p xml:space="preserve">This is <b>some</b> content.</></p>`.

**Comment by chrisbottin** [comment]
Fixed in version `3.4.1` by setting the new `strictMode` option to `true`.
